pennylane/devices/qubit/apply_operation.py

```diff
@@ -228,9 +229,27 @@ def _(op: type_op, state):
         [1., 0.]], requires_grad=True)
 
     """
+    if sp.sparse.issparse(op.matrix()):
+        return apply_operation_csr_matrix(op, state, is_state_batched)
```

Comment:
@AmintorDusko Here we ensured that op.matrix is always sparse.

Reply:
Nice, Thanks!